core/dbt/contracts/graph/semantic_manifest.py

```diff
@@ -58,6 +60,20 @@ def validate(self) -> bool:
         semantic_manifest = self._get_pydantic_semantic_manifest()
         validator = SemanticManifestValidator[PydanticSemanticManifest]()
         validation_results = validator.validate_semantic_manifest(semantic_manifest)
+        new_time_spines = semantic_manifest.project_configuration.time_spines
+        old_time_spines = semantic_manifest.project_configuration.time_spine_table_configurations
+        # If the new time spine contains a day grain then it is functionally equivalent to the legacy time spine.
+        new_time_spines_contain_day = any(
+            c for c in new_time_spines if c.primary_column.time_granularity == TimeGranularity.DAY
+        )
+        if (
+            get_flags().allow_mf_time_spines_without_yaml_configuration is False
+            and old_time_spines
+            and not new_time_spines_contain_day
+        ):
+            deprecations.warn(
+                "mf-timespine-without-yaml-configuration",
+            )
```

Comment:
Presumably this warning would not be terribly noisy -- just one per invocation?

Reply:
Yes, this is validation for the semantic manifest which there is only one of per project. I can't think of a way that this would fire more than once per invocation.